src/admin/hooks/useHotkey.tsx

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable no-shadow */
+import _ from 'lodash';
+import { useCallback, useEffect } from 'react';
+
+// Required to be outside of hook, else debounce would be necessary
+// and then one could not prevent the default behaviour.
+const pressedKeys = new Set([]);
+
+const map = {
+  MetaLeft: 'meta',
+  MetaRight: 'meta',
+  OSLeft: 'meta',
+  OSRight: 'meta',
+  ShiftLeft: 'shift',
+  ShiftRight: 'shift',
+  CtrlLeft: 'ctrl',
+  CtrlRight: 'ctrl',
+  AltLeft: 'alt',
+  AltRight: 'alt',
+  Escape: 'esc',
+};
+
+const stripKey = (key: string) => {
+  return (map[key] || key).trim()
+    .toLowerCase()
+    .replace('key', '');
+};
+
+const pushToKeys = (code: string) => {
+  const key = stripKey(code);
+
+  // There is a weird bug with macos that if the keys are not cleared they remain in the
+  // pressed keys set.
+  if (key === 'meta') {
+    pressedKeys.forEach((pressedKey) => pressedKey !== 'meta' && pressedKeys.delete(pressedKey));
+  }
+
+  pressedKeys.add(key);
+};
+
+const removeFromKeys = (code: string) => {
+  const key = stripKey(code);
+  // There is a weird bug with macos that if the keys are not cleared they remain in the
+  // pressed keys set.
+  if (key === 'meta') {
+    pressedKeys.clear();
+  }
+  pressedKeys.delete(key);
+};
+
+/**
+ * Hook function to work with hotkeys.
+ * @param param0.keyCode {string[]} The keys to listen for (`Event.code` without `'Key'` and lowercased)
+ * @param param0.cmdCtrlKey {boolean} Whether Ctrl on windows or Cmd on mac must be pressed
+ * @param func The callback function
+ */
+const useHotkey = (options: {
+  keyCodes: string[]
+  cmdCtrlKey: boolean
+}, func: (e, deps: any[]) => void, deps: any[]): void => {
+  const { keyCodes, cmdCtrlKey } = options;
+
+  const keydown = useCallback((e: KeyboardEvent) => {
+    if (e.key === undefined) {
+      // Autofill events, or other synthetic events, can be ignored
+      return;
+    }
+    pushToKeys(e.code);
+
+    // Check for Mac and iPad
+    const hasCmd = window.navigator.userAgent.includes('Mac OS X');
+    const pressedWithoutMeta = [...pressedKeys].filter((key) => key !== 'meta');
+
+    // Check whether arrays contain the same values (regardless of number of occurrences)
+    if (
+      _.isEmpty(_.xor(pressedWithoutMeta, keyCodes))
```

Comment:
We should be able to replace this bit of code with the following. This would allow us to replace `keyCodes` with a Set.
```
const setsAreEqual = <T>(lhs: Set<T>, rhs: Set<T>) => {
  return lhs.size === rhs.size && Array.from(lhs).every((value) => rhs.has(value));
};
```